src/app/spans.rs

```diff
@@ -0,0 +1,106 @@
+use crate::trace_context;
+use linkerd2_error::Error;
+use opencensus_proto::trace::v1 as oc;
+use std::collections::HashMap;
+use std::{error, fmt};
+use tokio::sync::mpsc;
+
+const SPAN_KIND_SERVER: i32 = 1;
+const SPAN_KIND_CLIENT: i32 = 2;
+
+#[derive(Clone)]
+pub struct SpanConverter {
```

Comment:
It would be good to have a comment explaining what, exactly, this thing actually does...

Reply:
why, it converts spans, of course!

Reply:
I don't understand why this is necessary. Why do we have to convert spans? Should this logic be part of linkerd-opencensus? What are we bridging that is best done in `app`?

Reply:
`linkerd-opencensus` and `linkerd-trace-context` are independent crates which do not depend on each other.  each has their own Span struct, so something which depends on both needs to convert from one to the other.  In order for this conversion to happen in `linkerd-opencensus`, it would need to depend on `linkerd-trace-context`.

Reply:
@adleong ahh ok. is the intent that that trace-context can be used with non-opencensus spans? i'll dig in more to read the details, but curious on how you think about the relationship between the components (i.e. where is decoupling important, etc)

Reply:
loose coupling is always a nice property to have, but realistically the crates are set up this way because it was the most expedient way to develop them in parallel.  will we ever replace opencensus with a different exporter? maybe.  but I don't want to over optimize for abstraction when we don't know what we want to abstract over.  

having self-contained library crates which are all tied together by the main crate is an architecture that makes sense to me.